Epoch reports error when compile succeeds

Upon successfully compiling an LF program, Epoch displays an error in the Error Log.  Requesting to display the error in the console gives this stack trace:
```
org.eclipse.core.commands.ExecutionException: Error executing 'org.lflang.lf.compile.toolbar': java.lang.NullPointerException: Cannot invoke "org.eclipse.core.resources.IFile.exists()" because "file" is null
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:170)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:485)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4604)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1524)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1547)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1532)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1325)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4390)
...
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.core.resources.IFile.exists()" because "file" is null
	at org.lflang.ui.actions.CompileActionHandler.collect(CompileActionHandler.java:293)
	at org.lflang.ui.actions.CompileActionHandler.execute(CompileActionHandler.java:145)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:283)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:97)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 40 more
```

--- a/org.lflang.ui/src/org/lflang/ui/actions/CompileActionHandler.java
+++ b/org.lflang.ui/src/org/lflang/ui/actions/CompileActionHandler.java
@@ -134,15 +134,18 @@ class CompileActionHandler extends AbstractHandler {
             var xtextEditor = EditorUtils.getActiveXtextEditor(event);
             if (xtextEditor != null) {
                 var xtextDocument = xtextEditor.getDocument();
+                var resource = xtextEditor.getResource();
                 if (xtextDocument != null) {
                     // Save editor
                     if (xtextEditor.isDirty()) {
                         xtextEditor.doSave(new NullProgressMonitor());
+                    } else if (resource != null) {
+                        // Load the resource of this editor based on its associated file (collect).
+                        // This does not retrieve the model resource from the editor directly but creates a new one. 
+                        // => workaround for issue #746.
+                        lfFiles.addAll(collect((IFile) resource));
                     }
-                    // Load the resource of this editor based on its associated file (collect).
-                    // This does not retrieve the model resource from the editor directly but creates a new one. 
-                    // => workaround for issue #746.
-                    lfFiles.addAll(collect((IFile) xtextEditor.getResource()));
+                    
                 }
             }
         } else if (selection instanceof IStructuredSelection) { // Invoked from context menu in project explorer

--- a/org.lflang.ui/src/org/lflang/ui/actions/CompileActionHandler.java
+++ b/org.lflang.ui/src/org/lflang/ui/actions/CompileActionHandler.java
@@ -139,7 +139,8 @@ class CompileActionHandler extends AbstractHandler {
                     // Save editor
                     if (xtextEditor.isDirty()) {
                         xtextEditor.doSave(new NullProgressMonitor());
-                    } else if (resource != null) {
+                    }
+                    if (resource != null) {
                         // Load the resource of this editor based on its associated file (collect).
                         // This does not retrieve the model resource from the editor directly but creates a new one. 
                         // => workaround for issue #746.

